### ISSUE TITLE
label中后四位geo表示啥意思？

### ISSUE THREAD

**Issue by BigPandaCPU** [huoyijie/AdvancedEAST#69]
hi,huoyijie，在label的时候，有个地方不太理解，就是这里的后四位，表示的是每一个点和最长边的点的距离吗？
 gt[i, j, 3:5] = xy_list[vs[long_edge][ith][0]] - [px, py]

 gt[i, j, 5:] = xy_list[vs[long_edge][ith][1]] - [px, py]

这里的xy_list[]记录的是这个4个点，按顺时针顺序存储的

这里的后四位gt[i,j,3:7]存储的表示 [px,py]这个点到这两个长边点的距离吗？为什么要计算这个距离呢？

**Comment by BigPandaCPU** [comment]
我调试了发现，这里的gt[i,j,3:]表示的是点[px,py]到xy_list[] 中离它最近两个点之间的坐标差值，在East原文中，说这里的gt[i,j,3:]表示的是点[px,py]到xy_list最小外接矩形的四条边的距离，这里作者能给解释一下为什么吗？
谢谢！

**Comment by BigPandaCPU** [comment]
![image](https://user-images.githubusercontent.com/25834151/55309537-b29a6f00-5490-11e9-9711-5fe88a85578b.png)

**Comment by BigPandaCPU** [comment]
今天通过对源代码的解读搞清楚了，与大家分享一下
label说明：（这里是以long_edge为上下边为例，若long_edge为左右边时，将该图顺时针旋转90度就是）
黑色框：表示标注的文本框
蓝色框：表示对原始文本框的四边，缩小0.2倍后的线框（是以短边的0.2）
红色框：表示对原始文本框的长边缩小0.6位后的线框（是以短边的0.6倍）
黄色区域：标注的是文本头
绿色区域：标注的是文本尾

在gt[i,j,3：]表示的属于 属于头部和尾部的点离它最近的两个点的坐标差
如下图：p点在头部，则这里的gt[i,j,3:]表示的是点p到0‘ 和1‘之间的坐标差

![image](https://user-images.githubusercontent.com/25834151/55310522-30f81080-5493-11e9-8e5c-0bb5cf5fc3b9.png)

**Comment by snowwindy** [comment]
你好，谢谢您的解释，但是我对这一部分的损失函数不是特别懂，能帮忙解释一下吗？

**Comment by BigPandaCPU** [comment]
这里的loss计算分成三部分：
第一部分是计算是否是文本，就是预测的第1位，用的是交叉熵；
第二部分是计算是边界像素，就是预测的第2,3位，用的也是交叉熵；
第三部分是计算预测的四个端点坐标之间，就是预测的第4,5,6,7，用的Smooth_L1_loss。
![image](https://user-images.githubusercontent.com/25834151/56256551-c0d8c400-60fb-11e9-956f-32e7744287a8.png)

**Comment by snowwindy** [comment]
嗯嗯，好的，谢谢啦

**Comment by BigPandaCPU** [comment]
已经搞清楚了这个问题了

**Comment by snowwindy** [comment]
你好，想再问一下，后四位是其中的像素点到真实标签顶点的坐标差，还是到缩小框顶点的坐标差

**Comment by BigPandaCPU** [comment]
真实标签顶点的坐标差。最后预测的框的坐标就是直接将预测的顶点坐标取加权平均后得到的




------------------ 原始邮件 ------------------
发件人: "snowwindy"<notifications@github.com>;
发送时间: 2019年5月5日(星期天) 晚上7:49
收件人: "huoyijie/AdvancedEAST"<AdvancedEAST@noreply.github.com>;
抄送: "石油大学-汪雄"<1196449387@qq.com>; "State change"<state_change@noreply.github.com>; 
主题: Re: [huoyijie/AdvancedEAST] label中后四位geo表示啥意思？ (#69)




你好，想再问一下，后四位是其中的像素点到真实标签顶点的坐标差，还是到缩小框顶点的坐标差

**Comment by hxk11111** [comment]
@huoyijie @BigPandaCPU 你好，想问一下，如何理解损失函数中的vertex_beta，
```
side_vertex_code_loss = \
        tf.reduce_sum(tf.reduce_sum(pos + neg, axis=-1) * positive_weights) / (
                tf.reduce_sum(positive_weights) + cfg.epsilon)
```
这一段要怎么理解呢。

我想多回归两个点，即文本行中心的上边界点和下边界点，要如何修改损失函数呢

**Comment by Andiez-Nguyen** [comment]
> 这里的loss计算分成三部分：
> 第一部分是计算是否是文本，就是预测的第1位，用的是交叉熵；
> 第二部分是计算是边界像素，就是预测的第2,3位，用的也是交叉熵；
> 第三部分是计算预测的四个端点坐标之间，就是预测的第4,5,6,7，用的Smooth_L1_loss。
> ![image](https://user-images.githubusercontent.com/25834151/56256551-c0d8c400-60fb-11e9-956f-32e7744287a8.png)

Thanks for your explaination. But I have some confusions in loss for side vertex code and loss for side_vertex_coor delta. Can you explain it or share any ref for me? Thanks a lot

**Comment by xiiiiiiii** [comment]
> @huoyijie @BigPandaCPU 你好，想问一下，如何理解损失函数中的vertex_beta，
> 
> ```
> side_vertex_code_loss = \
>         tf.reduce_sum(tf.reduce_sum(pos + neg, axis=-1) * positive_weights) / (
>                 tf.reduce_sum(positive_weights) + cfg.epsilon)
> ```
> 
> 这一段要怎么理解呢。
> 
> 我想多回归两个点，即文本行中心的上边界点和下边界点，要如何修改损失函数呢

你最后解决这个问题了吗？我的想法是即使多回归两个点，只需要在标签上做修改即可，损失函数不用变
